### ISSUE TITLE
Mapping terminated before handling trigger: oneshot canceled

### ISSUE THREAD

**Issue by ksvirsky** [graphprotocol/graph-node#2113]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
Hosted graph service is failed to sync on rinkeby with error:
Subgraph instance failed to run: Failed to process trigger in block #7895457 (4d45a5d05ad08a6cafe9f855769ddff27dd2d293bf640cc71e79cd068b441056), transaction f77f828af1ccf48f55c5d9040e231504e7023237d1151cb1ec346ca8a36bebe3: Mapping terminated before handling trigger: oneshot canceled, code: SubgraphSyncingFailure, id: QmdzPa9266GXNYfLcABs99BuWSUvo2hjWHi25yQa1fvx3i

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
All the code with contracts and graph mapping:
https://github.com/ksvirsky/graph-oneshot-error
Deployed subgraph:
https://thegraph.com/explorer/subgraph/ksvirsky/graph-oneshot-error?selected=logs

In short:
1. Factory emits event with new contract address.
2. Factory subgraph mapping handles it and create template.
3. Templated contract emits event.
4. Template subgraph mapping handles emitted event and binds contract instance.
5. Calling any method on bound contract causes the error.

**What is the expected behavior?**
If the issue is in mapping/configuration, then the error message is confused/useless and should be improved. Otherwise it looks like the issue is in internal code.

**Comment by ksvirsky** [comment]
For anyone having the same issue for templates, check if you have "**apiVersion: 0.0.4**" for template section in subgraph.yaml.

**Comment by techstreamp** [comment]
Is this issue solved?

**Comment by leoyvens** [comment]
@techstreamp ping me on our Discord at `#subgraph-development` and I can try to help.

**Comment by nhynhyiddy** [comment]
I also encounter the same bug on creating template.

**Comment by milonite** [comment]
There's any update on this? I'm encountering the same issue
